ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Overview/CVEsTable.tsx

```diff
@@ -1,48 +1,230 @@
+            distroTuples {
+                summary
+                operatingSystem
+                cvss
+                scoreVersion
+            }
+        }
+    }
+`;
+
+const totalNodeCountQuery = gql`
+    query getTotalNodeCount {
+        nodeCount
+    }
+`;
+
+// TODO Need to verify these types with the BE implementation
+export type NodeCVE = {
+    cve: string;
+    nodeCountBySeverity: {
+        critical: { total: number };
+        important: { total: number };
+        moderate: { total: number };
+        low: { total: number };
+    };
+    topCVSS: number;
+    affectedNodeCount: number;
+    firstDiscoveredInSystem: string;
+    distroTuples: {
+        summary: string;
+        operatingSystem: string;
+        cvss: number;
+        scoreVersion: string;
+    }[];
+};
+
 export type CVEsTableProps = {
     querySearchFilter: QuerySearchFilter;
     isFiltered: boolean;
     pagination: ReturnType<typeof useURLPagination>;
 };
 
-function CVEsTable({
-    /* eslint-disable @typescript-eslint/no-unused-vars */
-    querySearchFilter,
-    isFiltered,
-    pagination,
-    /* eslint-enable @typescript-eslint/no-unused-vars */
-}: CVEsTableProps) {
-    // TODO - Placeholders for query results
-    const data = [];
-    const previousData = undefined;
-    const error: Error | undefined = undefined;
-    const loading = false;
+function CVEsTable({ querySearchFilter, isFiltered, pagination }: CVEsTableProps) {
+    const { page, perPage } = pagination;
+    const { data, previousData, loading, error } = useQuery<
+        { nodeCVEs: NodeCVE[] },
+        {
+            query: string;
+            pagination: {
+                offset: number;
+                limit: number;
+            };
+        }
+    >(cvesListQuery, {
+        variables: {
+            query: getRegexScopedQueryString(querySearchFilter),
+            pagination: {
+                offset: (page - 1) * perPage,
+                limit: perPage,
+            },
+        },
+    });
+
+    const totalNodeCountRequest = useQuery<{ nodeCount: number }>(totalNodeCountQuery);
+    const totalNodeCount = totalNodeCountRequest.data?.nodeCount ?? 0;
 
     const tableData = data ?? previousData;
+    const tableState = getTableUIState({
+        isLoading: loading,
+        data: tableData?.nodeCVEs,
+        error,
+        searchFilter: querySearchFilter,
+    });
+
+    const expandedRowSet = useSet<string>();
+    const colSpan = 6;
 
     return (
-        <>
-            {loading && !tableData && (
-                <Bullseye>
-                    <Spinner />
-                </Bullseye>
-            )}
-            {error && (
-                <TableErrorComponent error={error} message="Adjust your filters and try again" />
+        <Table
+            borders={tableState.type === 'COMPLETE'}
+            variant="compact"
+            role="region"
+            aria-live="polite"
+            aria-busy={loading ? 'true' : 'false'}
```

Comment:
aside: I'm surprised these aren't the default in PatternFly. It may be worth making a Feature Request to their project.

Reply:
So conceptually I think this is a good idea, but in practice I wasn't able to get results that I was happy with when using a screen reader. I'm not entirely sure it is an issue with the pattern though, and is more my inexperience in navigating a page with a screen reader.

I can make a note to bring this up at one of the PF meetings though since it is more likely they have someone with better expertise on their team.